cpp/include/cudf/table/primitive_row_operators.cuh

```diff
@@ -126,21 +141,19 @@ class row_equality_comparator {
    */
   __device__ bool operator()(size_type lhs_row_index, size_type rhs_row_index) const
   {
-    if (_has_nulls) {
-      bool const lhs_is_null{_lhs.column(0).is_null(lhs_row_index)};
-      bool const rhs_is_null{_rhs.column(0).is_null(rhs_row_index)};
-      if (lhs_is_null and rhs_is_null) {
-        return _nulls_are_equal == null_equality::EQUAL;
-      } else if (lhs_is_null != rhs_is_null) {
+    for (size_type i = 0; i < _lhs.num_columns(); ++i) {
```

Comment:
I intentionally moved null handling from the element level to the row level, as it's independent of the column type. Passing null information through the type dispatcher feels redundant and is likely to increase register usage due to reference passing, which might explain the performance slowdown in some cases. I could be wrong though.

Reply:
There is actually a decrease in register usage(I dont know why yet). But this does not affect occupancy

Reply:
moved back to row level